ext/openssl/ossl_engine.c

```diff
@@ -46,13 +52,25 @@ VALUE eEngineError;
 /*
  * Private
  */
+#if !defined(LIBRESSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x10100000
+#define OSSL_ENGINE_LOAD_IF_MATCH(x) \
+do{\
+  if(!strcasecmp(#x, RSTRING_PTR(name))){\
+    if (OPENSSL_init_crypto(OPENSSL_INIT_ENGINE_##x, NULL))\
+      return Qtrue;\
+    else\
+      rb_raise(rb_eRuntimeError, "OPENSSL_init_ssl");\
+  }\
+}while(0)
+#else
 #define OSSL_ENGINE_LOAD_IF_MATCH(x) \
 do{\
   if(!strcmp(#x, RSTRING_PTR(name))){\
     ENGINE_load_##x();\
     return Qtrue;\
   }\
 }while(0)
+#endif
```

Comment:
strcasecmp() would be overkill. I was expecting something like:

```
#define OSSL_ENGINE_LOAD_IF_MATCH(engine_name, x) do { \
  ...
  if (!strcmp(#engine_name, RSTRING_PTR(name)))...
...

  OSSL_ENGINE_LOAD_IF_MATCH(dynamic, DYNAMIC);
```

The OpenSSL < 1.1.0 version can have the same signature so the #ifs below can be removed.

Additionally, (this is not documented anywhere, but) ossl_raise() that rewinds "error"s in the OpenSSL error queue should be used with eEngineError instead. ossl.c had to use the raw rb_raise() because OpenSSL wouldn't be initialized at that time.